File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/effect/UXInnerShadowEffect.java
package org.develnext.jphp.ext.javafx.classes.effect;

import javafx.scene.effect.BlurType;
import javafx.scene.effect.DropShadow;
import javafx.scene.effect.Effect;
import javafx.scene.effect.InnerShadow;
import javafx.scene.paint.Color;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.*;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "effect\\UXInnerShadowEffect")
public class UXInnerShadowEffect extends UXEffect<InnerShadow> {
    interface WrappedInterface {
        @Property double radius();
        @Property Color color();
        @Property double offsetX();
        @Property double offsetY();

        @Property double height();
        @Property double width();

        @Property BlurType blurType();

        @Property Effect input();
    }

    public UXInnerShadowEffect(Environment env, InnerShadow wrappedObject) {
        super(env, wrappedObject);
    }

    public UXInnerShadowEffect(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new InnerShadow();
    }

    @Signature
    public void __construct(double radius) {
        __wrappedObject = new InnerShadow();
        getWrappedObject().setRadius(radius);
    }

    @Signature
    public void __construct(double radius, Color color) {
        __wrappedObject = new InnerShadow(radius, color);
    }

    @Signature
    public void __construct(double radius, Color color, double offsetX, double offsetY) {
        __wrappedObject = new InnerShadow(radius, offsetX, offsetY, color);
    }

    @Setter
    public void setSize(double[] size) {
        if (size.length >= 2) {
            getWrappedObject().setWidth(size[0]);
            getWrappedObject().setHeight(size[1]);
        }
    }

    @Getter
    public double[] getSize() {
        return new double[] { getWrappedObject().getWidth(), getWrappedObject().getHeight() };
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/effect/UXInnerShadowEffect.php
<?php
namespace php\gui\effect;

use php\gui\paint\UXColor;

/**
 * Class UXInnerShadowEffect
 * @package php\gui\effect
 * @packages gui, javafx
 */
class UXInnerShadowEffect extends UXEffect
{
    /**
     * @var string ONE_PASS_BOX, TWO_PASS_BOX, THREE_PASS_BOX, GAUSSIAN
     */
    public $blurType = 'THREE_PASS_BOX';

    /**
     * @var UXColor
     */
    public $color;

    /**
     * @var double
     */
    public $radius;

    /**
     * @var float
     */
    public $offsetX = 0.0;

    /**
     * @var float
     */
    public $offsetY = 0.0;

    /**
     * @var double
     */
    public $width;

    /**
     * @var double
     */
    public $height;

    /**
     * @var float[] width + height
     */
    public $size = [0.0, 0.0];

    /**
     * @param double $radius (optional)
     * @param UXColor|string $color (optional)
     * @param double $offsetX (optional)
     * @param double $offsetY (optional)
     */
    public function __construct($radius, $color, $offsetX, $offsetY)
    {
    }
}